### PR TITLE
Workaround unprivileged PacketConn on darwin returns the entire IP packet in icmp.ListenPacket()

### PR DESCRIPTION
**Pull request by floatingstatic**
TLDR: unprivileged icmp echo replies are discarded on macOS (darwin)

This is a patch to work around an issue described in https://github.com/golang/go/issues/47369 which can be summarized as follows:

Due to [setting the IP_STRPHDR socket option](https://cs.opensource.google/go/x/net/+/refs/tags/v0.10.0:icmp/listen_posix.go;l=74-75;bpv=0;bpt=0) in darwin  (macOS) environments, 
reads on unprivileged sockets (udp) include the IP header before the ICMP header in the read bytes returned by `icmp.ListenPacket()`.  As a result of this, `pro-bing` ends up [discarding](https://github.com/prometheus-community/pro-bing/blob/a17ba032a911c7afdd16c0ef39278e603f2614b8/ping.go#L662-L664) icmp echo replies due to the unexpected ip header bytes returned by `icmp.ListenPacket()`.

While this problem needs to be addressed in go itself, this patch provides a work around until this issue is addressed.

### CODE REVIEW

**Comment by floatingstatic** [comment]
Also worthing pointing out `pinger.SetPrivileged(true)` works fine on macOS, this is only a problem when this is `false`

**Review by SuperQ — APPROVED** [review]
Thanks!
